Hermes/appscale/hermes/stats/stats_app.py

```diff
@@ -144,6 +154,143 @@ def get_cluster_stats_api_routes(is_master):
   ]
 
 
+class ProfilingManager(object):
+  """
+  This manager watches stats profiling configs in Zookeeper,
+  when configs are changed it starts/stops/restarts periodical
+  tasks which writes profile log with proper parameters.
+  """
+
+  def __init__(self, zk_client):
+    """ Initializes instance of ProfilingManager.
+    Starts watching profiling configs in zookeeper.
+
+    Args:
+      zk_client: an instance of KazooClient - started zookeeper client.
+    """
+    self.nodes_profile_log = None
+    self.processes_profile_log = None
+    self.proxies_profile_log = None
+    self.nodes_profile_task = None
+    self.processes_profile_task = None
+    self.proxies_profile_task = None
+
+    def bridge_to_ioloop(update_function):
+      """ Creates function which schedule execution of update_function
+      inside current IOLoop.
+
+      Args:
+        update_function: a function to execute in IOLoop.
+      Returns:
+        A callable which schedules execution of update_function inside IOLoop.
+      """
+      def update_in_ioloop(new_conf, znode_stat):
+        update = lambda: update_function(new_conf, znode_stat)
```

Comment:
No need to change it now, but just for future reference, `add_callback` accepts arguments so the lambda is not necessary.